query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -94,6 +94,8 @@ public class LifecycleManager implements ModuleLifecycle {
     */
    public static final String MAX_BOOLEAN_CLAUSES_SYS_PROP = "infinispan.query.lucene.max-boolean-clauses";
 
+   public static boolean REMOTE_QUERY_MODULE_IS_PRESENT = false;
```

Comment:
I don't like non-final `public static` fields which are directly modifiable. Maybe making it an `AtomicBoolean` or adding a `public static void enableRemoteQuery()` method that sets the field to true would be better.

Reply:
I changed the way to set the flag, using the {{GlobalComponentRegistry}} to lookup the query module from the remote query module. So that we can use a non-static field.